File: Games/Kevin/src/de/hsh/kevin/logic/Leben.java
package de.hsh.kevin.logic;

/**
 * Leben erstellt ein Counter
 * @author Kevin
 *
 */
public class Leben {
    private int leben;
    
    /**
     * Erstellt das Objekt mit der Aktuellen anzahl Leben, die in SkinConfig.getLife() liefert
     */
    public Leben() {
	leben = (int) Config.getLife();
    }
    
    /**
     * Erhöht counter um 1
     */
    public void increase() {
	leben++;
    }
    
    /**
     * Verringert counter um 1
     */
    public void decrease() {
	leben--;
    }
    
    /**
     *  Liefert Anzahl der Leben
     * @return Anzahl der Leben
     */
    public int getLeben() {
	return leben;
    }

}

File: Games/Kevin/src/de/hsh/kevin/logic/Leben.kt
package de.hsh.kevin.logic

private const val maxLife = 4

/**
 * Liefert Faktor für die Leben je nach Schwierigkeit, mit easy als Standard und andere mit niedrigem Wert
 * @return
 */
private val lifeFactor: Double
    get() = when (Config.difficultyOption) {
        enmDifficultyOptions.easy   -> 1.0
        enmDifficultyOptions.normal -> 0.75
        enmDifficultyOptions.hard   -> 0.5
    }

/**
 * Leben erstellt ein Counter
 * @author Kevin
 */
data class Leben(var leben: Int = (lifeFactor * maxLife).toInt()) {
    /**
     * Verringert counter um 1
     */
    fun decrease() {
        leben--
    }



}
